src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,160 @@
+import cn from 'classnames';
+import {
+  useState,
+  useContext,
+  FormEvent,
+} from 'react';
+import { TodosContext } from '../../TodosContext';
+import { Todo } from '../../types';
+
+interface Props {
+  todo: Todo,
+  isTempTodo?: boolean,
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  isTempTodo,
+}) => {
+  const {
+    todoDelete,
+    areCompletedDeletingNow,
+    todoUpdate,
+    toggleToStatus,
+  } = useContext(TodosContext);
+
+  const [isEditing, setIsEditing] = useState(false);
+  const [query, setQuery] = useState(todo.title);
+
+  const [isLoading, setIsLoading] = useState(isTempTodo || false);
+
+  const isChangingStatus = typeof toggleToStatus === 'boolean'
+    && todo.completed !== toggleToStatus;
+
+  const handleOnToggle = () => {
+    const newTodo: Todo = {
+      ...todo,
+      completed: !todo.completed,
+    };
```

Comment:
according to checklist, you should use destructurization everywhere you can

Reply:
Can you please explain what you mean in this case? Where should I use destructurization here?

Reply:
Sure, there is a checklist file where you can find this point -Use destructuring wherever possible. It makes code more readable.
You should destructure the todo object with just the title, completed, and id.
I hope I explained well)

Reply:
Yes, I understand the part about the checklist but here I need to change the first object's value so I don't understand why I need here destructuring assignment. Or do you mean to change `todo.title` and `todo.completed` to destructured variables?

Reply:
Yes, I mean instead of using `todo.completed` or `todo.title` you should use destructured variables from todo object. And even if you are changing object's value, instead of `todo.completed` it will be just `completed` from `const { title, completed, id } = todo;`